### ISSUE TITLE
Using dlib to make alexnet work on mnist dataset,and the loss value is always up and down 2.305

### ISSUE THREAD

**Issue by whiteandbright** [davisking/dlib#2006]
Using dlib to make alexnet work on mnist dataset,but the loss value is alwys floating up and down at 2.305.it seems like if the net include too much parameters,the loss is hard to reduce. 

## Expected Behavior
Training should reduce the model's loss value and eventually converge to a relatively small value .

## Current Behavior
I use the dlib to build up alexnet ,and run the alexnet on the mnist dataset.

1.First I use the dnn_introduction_ex.cpp as model ,this cpp file use LeNet5 as an example,so I simply replace the net_type to build up alexnet,this is the only thing I do,and training loss is alwys floating up and down at 2.305.and no matter how you change the loss value, the phenomenon is the same, the loss value has been floating up and down at 2.305.

2.if I reduce the conv layer filter kernal number,it will coverage,for example,I change the third conv layer filter number from 384 to 20,this net will worked.also if I reduce the last two fc layer from 4096 node to 1000 node it also worked.

## Steps to Reproduce
you can reproduce this problem very easily.copy the dnn_introduction_ex.cpp,and then change the net_type to   
```
using net_type =
  loss_multiclass_log<fc<10, 
       dropout<relu<fc<4096, 
       dropout<relu<fc<4096, 
       max_pool<3, 3, 2, 2, relu<con<256, 3, 3, 1, 1,  
       relu<con<384, 3, 3, 1, 1, 
       relu<con<384, 3, 3, 1, 1,
       max_pool<3, 3, 2, 2, l2normalize<relu<con<256, 5, 5, 1, 1, 
       max_pool<3, 3, 2, 2, l2normalize<relu<con<96, 11, 11, 1, 1, 
       input<matrix<unsigned char>>>>>>>>>>>>>>>>>>>>>>>>>;
```
and no dnn_introduction_ex.cpp other code need  to be changed.  

* **Version**: 1.19.19
* **Where did you get dlib**: official source ,I get it from dlib.net
* **Platform**: windows10 64bit
* **Compiler**: using visual studio 2015

**Comment by davisking** [comment]
Maybe you need a smaller learning rate.

**Comment by whiteandbright** [comment]
I have changed a lot of original learning rates such as 0.0001, 0.001, 0.01, 0.05, 0.1.the result are the same,the loss seems like never reducing

**Comment by davisking** [comment]
Those l2normalize layers don't really make any sense and are probably
messing this up.
